### PR TITLE
Add OCLC Selector and Record

### PR DESCRIPTION
**Pull request by maxkadel**
Connected to #234

### CODE REVIEW

File: app/models/oclc/record.rb [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      return nil if targets.empty?
+
+      targets.map(&:value).join(' ')
+    end
+
+    # Only the class from the call_number
+    def lc_class
+      return unless call_number
+
+      call_number.gsub(/^([A-Z]+?)[^A-Z].*$/, '\1')
```
Comment (ishasinha1) [inline app/models/oclc/record.rb:137]: Jane and I are a little confused about this regular expression. Could you please explain what this is supposed to do?

File: spec/models/oclc/record_spec.rb [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+require 'rails_helper'
+
+RSpec.describe Oclc::Record, type: :model do
+  let(:marc_record) { marc_reader.first }
+  let(:marc_reader) { MARC::Reader.new(oclc_fixture_file_path.to_s) }
+  let(:selector_config) { Rails.application.config.newly_cataloged.selectors.first }
+  let(:selector) { Oclc::Selector.new(selector_config:) }
+  let(:subject) { described_class.new(marc_record:) }
+
+  context 'with a newer fixture' do
```
Comment (ishasinha1) [inline spec/models/oclc/record_spec.rb:11]: What's the difference between the newer one and the older one?

File: spec/models/oclc/record_spec.rb [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        expect(subject.juvenile?).to eq(false)
+        expect(subject.audiobook?).to eq(false)
+        expect(subject.published_in_us_uk_or_canada?).to eq(false)
+        expect(subject.monograph?).to eq(true)
+        expect(subject.within_last_two_years?).to eq(true)
+        expect(subject.generally_relevant?).to eq(true)
+        expect(subject.relevant_to_selector?(selector:)).to eq(true)
+      end
+
+      # xit 'can go through all the records to find a relevant one' do
```
Comment (ishasinha1) [inline spec/models/oclc/record_spec.rb:29]: Is this still needed?

File: app/models/oclc/record.rb [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    end
+
+    def author
+      # The first available author field
+      auth_field = record.fields(%w[100 110 111]).try(:[], 0)
+      return '' if auth_field.blank?
+
+      auth_tag = auth_field.tag
+      subf_to_skip = auth_subfields_to_skip(auth_tag)
+      targets = auth_field.subfields.reject do |subfield|
```
Comment (ishasinha1) [inline app/models/oclc/record.rb:48]: Could you rename this to 'subf_to_keep'?

File: app/models/oclc/record.rb [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+        lccn << StdNum::LCCN.normalize(field.value)
+      end
+      lccn.uniq.join(' | ')
+    end
+
+    def pub_field
+      return record['260'] if record['260']
+
+      record.fields('264').min_by(&:indicator2) if record.fields('264').present?
```
Comment (ishasinha1) [inline app/models/oclc/record.rb:92]: This is a really cool use of min_by!!

File: app/models/oclc/record.rb [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        subf_to_skip.include?(subfield.code)
+      end
+
+      targets.map { |field| scrub_string(field.value) }.join(' ')
+    end
+
+    def description
+      return '' unless record['300']
+
+      record['300'].subfields.map(&:value).join(' ')
```
Comment (ishasinha1) [inline app/models/oclc/record.rb:58]: Would record['300'].to_s accomplish a similar purpose?

File: app/models/oclc/record.rb (hunk 1 above)
Comment (ishasinha1) [reply app/models/oclc/record.rb:137]: Also the regexp in lc_number! 😕

**Review by rladdusaw — APPROVED** [review]

File: app/models/oclc/record.rb [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+        %w[0 6 j]
+      end
+    end
+
+    def scrub_string(string)
+      return string if string.nil?
+
+      new_string = string.dup
+      new_string.strip!
+      new_string[-1] = '' if %r{[.,:/=]}.match?(new_string[-1])
```
Comment (ishasinha1) [inline app/models/oclc/record.rb:235]: We wanted to inform you that we understand this regular expression.

**Review by ishasinha1 — APPROVED** [review]
Just some things we noticed, none of them are blockers. Thank you @maxkadel!

File: app/models/oclc/record.rb (hunk 6 above)
Comment (maxkadel) [reply app/models/oclc/record.rb:58]: For one of test objects, the output of the above code is `"volumes <1, 2-3, 5-14>; 25 cm."`

The output of `record['300'].to_s` is `"300    $a volumes <1, 2-3, 5-14>; $c 25 cm."`, which I think does not serve our purpose of human readability in the context of a CSV.

File: app/models/oclc/record.rb (hunk 5 above)
Comment (maxkadel) [reply app/models/oclc/record.rb:92]: Thanks! But it was all @mzelesky!

File: spec/models/oclc/record_spec.rb (hunk 3 above)
Comment (maxkadel) [reply spec/models/oclc/record_spec.rb:29]: Nope - removed.

**Pull request merged**